[GTK][WPE] Add DRMDeviceManager to replace GBMDevice

carlosgcampos opened the pull request:
#### 9b709d7e78e85b3f1f8f4ccc21e4716fa9502925
<pre>
[GTK][WPE] Add DRMDeviceManager to replace GBMDevice
<a href="https://bugs.webkit.org/show_bug.cgi?id=272886">https://bugs.webkit.org/show_bug.cgi?id=272886</a>

Reviewed by Alejandro G. Castro.

GBMDevice assumes there&apos;s always a single GPU device, but linux dmabuf
feedback wayland protocol can give a different target device, so we need
a way to handle multiple devices.

* Source/WebCore/SourcesGTK.txt:
* Source/WebCore/SourcesWPE.txt:
* Source/WebCore/platform/TextureMapper.cmake:
* Source/WebCore/platform/graphics/PlatformDisplay.cpp:
(WebCore::PlatformDisplay::createPlatformDisplay):
(WebCore::PlatformDisplay::gbmDevice):
* Source/WebCore/platform/graphics/gbm/DRMDeviceManager.cpp: Added.
(WebCore::DRMDeviceManager::singleton):
(WebCore::DRMDeviceManager::initializeMainDevice):
(WebCore::DRMDeviceManager::mainDeviceNode const):
(WebCore::DRMDeviceManager::mainGBMDeviceNode const):
(WebCore::DRMDeviceManager::deviceNode):
* Source/WebCore/platform/graphics/gbm/DRMDeviceManager.h: Added.
(WebCore::DRMDeviceManager::isInitialized const):
* Source/WebCore/platform/graphics/gbm/DRMDeviceNode.cpp: Added.
(WebCore::DRMDeviceNode::create):
(WebCore::DRMDeviceNode::DRMDeviceNode):
(WebCore::DRMDeviceNode::~DRMDeviceNode):
(WebCore::DRMDeviceNode::gbmDevice const):
* Source/WebCore/platform/graphics/gbm/DRMDeviceNode.h: Added.
(WebCore::DRMDeviceNode::filename const):
* Source/WebCore/platform/graphics/gbm/GBMBufferSwapchain.cpp:
(WebCore::GBMBufferSwapchain::getBuffer):
* Source/WebCore/platform/graphics/gbm/GraphicsContextGLGBM.cpp:
(WebCore::GraphicsContextGLGBM::platformInitializeContext):
* Source/WebCore/platform/graphics/gstreamer/DMABufVideoSinkGStreamer.cpp:
(webKitDMABufVideoSinkIsEnabled):
* Source/WebKit/GPUProcess/GPUProcess.cpp:
(WebKit::GPUProcess::initializeGPUProcess):
* Source/WebKit/WebProcess/WebPage/dmabuf/AcceleratedSurfaceDMABuf.cpp:
(WebKit::AcceleratedSurfaceDMABuf::RenderTargetEGLImage::create):
* Source/WebKit/WebProcess/glib/WebProcessGLib.cpp:
(WebKit::WebProcess::platformInitializeWebProcess):

Canonical link: <a href="https://commits.webkit.org/277664@main">https://commits.webkit.org/277664@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/36d408a76cc5c394733e0bc20e1ed6590abbc50b

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/48263 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/27475 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/51217 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/50951 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/44328 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/50568 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/33409 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/24995 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/39422 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/48845 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/25171 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/41699 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/20571 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/22651 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/42879 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/6319 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/44609 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/43332 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/52855 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/23310 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/19661 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/46761 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/24575 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/41880 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/45674 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/25380 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/6853 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/24298 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

alexgcastro reviewed (approved):
LGTM! Really important for the situations with multiple devices.

carlosgcampos committed:
[GTK][WPE] Add DRMDeviceManager to replace GBMDevice
https://bugs.webkit.org/show_bug.cgi?id=272886

Reviewed by Alejandro G. Castro.

GBMDevice assumes there's always a single GPU device, but linux dmabuf
feedback wayland protocol can give a different target device, so we need
a way to handle multiple devices.

* Source/WebCore/SourcesGTK.txt:
* Source/WebCore/SourcesWPE.txt:
* Source/WebCore/platform/TextureMapper.cmake:
* Source/WebCore/platform/graphics/PlatformDisplay.cpp:
(WebCore::PlatformDisplay::createPlatformDisplay):
(WebCore::PlatformDisplay::gbmDevice):
* Source/WebCore/platform/graphics/gbm/DRMDeviceManager.cpp: Added.
(WebCore::DRMDeviceManager::singleton):
(WebCore::DRMDeviceManager::initializeMainDevice):
(WebCore::DRMDeviceManager::mainDeviceNode const):
(WebCore::DRMDeviceManager::mainGBMDeviceNode const):
(WebCore::DRMDeviceManager::deviceNode):
* Source/WebCore/platform/graphics/gbm/DRMDeviceManager.h: Added.
(WebCore::DRMDeviceManager::isInitialized const):
* Source/WebCore/platform/graphics/gbm/DRMDeviceNode.cpp: Added.
(WebCore::DRMDeviceNode::create):
(WebCore::DRMDeviceNode::DRMDeviceNode):
(WebCore::DRMDeviceNode::~DRMDeviceNode):
(WebCore::DRMDeviceNode::gbmDevice const):
* Source/WebCore/platform/graphics/gbm/DRMDeviceNode.h: Added.
(WebCore::DRMDeviceNode::filename const):
* Source/WebCore/platform/graphics/gbm/GBMBufferSwapchain.cpp:
(WebCore::GBMBufferSwapchain::getBuffer):
* Source/WebCore/platform/graphics/gbm/GraphicsContextGLGBM.cpp:
(WebCore::GraphicsContextGLGBM::platformInitializeContext):
* Source/WebCore/platform/graphics/gstreamer/DMABufVideoSinkGStreamer.cpp:
(webKitDMABufVideoSinkIsEnabled):
* Source/WebKit/GPUProcess/GPUProcess.cpp:
(WebKit::GPUProcess::initializeGPUProcess):
* Source/WebKit/WebProcess/WebPage/dmabuf/AcceleratedSurfaceDMABuf.cpp:
(WebKit::AcceleratedSurfaceDMABuf::RenderTargetEGLImage::create):
* Source/WebKit/WebProcess/glib/WebProcessGLib.cpp:
(WebKit::WebProcess::platformInitializeWebProcess):

Canonical link: https://commits.webkit.org/277664@main
